odl/trafos/backends/pyfftw_bindings.py

```diff
@@ -0,0 +1,309 @@
+           for flag in ('FFTW_MEASURE', 'FFTW_PATIENT', 'FFTW_EXHAUSTIVE',
+                        'FFTW_DESTROY_INPUT')):
+        return True
+    elif (direction in ('backward', 'FFTW_BACKWARD') and halfcomplex and
+          ndim != 1):
+        return True
+    else:
+        return False
+
+
+def _pyfftw_check_args(arr_in, arr_out, axes, halfcomplex, direction):
+    """Raise an error if anything is not ok with in and out."""
+    if len(set(axes)) != len(axes):
+        raise ValueError('duplicate axes are not allowed')
+
+    if direction == 'forward':
+        out_shape = list(arr_in.shape)
+        if halfcomplex:
+            try:
+                out_shape[axes[-1]] = arr_in.shape[axes[-1]] // 2 + 1
+            except IndexError as err:
+                raise_from(IndexError('axis index {} out of range for array '
+                                      'with {} axes'
+                                      ''.format(axes[-1], arr_in.ndim)),
+                           err)
+
+        if arr_out.shape != tuple(out_shape):
+            raise ValueError('expected output shape {}, got {}'
+                             ''.format(tuple(out_shape), arr_out.shape))
+
+        if is_real_dtype(arr_in.dtype):
+            out_dtype = TYPE_MAP_R2C[arr_in.dtype]
+        elif halfcomplex:
+            raise ValueError('cannot combine halfcomplex forward transform '
+                             'with complex input')
+        else:
+            out_dtype = arr_in.dtype
+
+        if arr_out.dtype != out_dtype:
+            raise ValueError('expected output dtype {}, got {}'
+                             ''.format(dtype_repr(out_dtype),
+                                       dtype_repr(arr_out.dtype)))
+
+    elif direction == 'backward':
+        in_shape = list(arr_out.shape)
+        if halfcomplex:
+            try:
+                in_shape[axes[-1]] = arr_out.shape[axes[-1]] // 2 + 1
+            except IndexError as err:
+                raise_from(IndexError('axis index {} out of range for array '
+                                      'with {} axes'
+                                      ''.format(axes[-1], arr_out.ndim)),
+                           err)
+
+        if arr_in.shape != tuple(in_shape):
+            raise ValueError('expected input shape {}, got {}'
+                             ''.format(tuple(in_shape), arr_in.shape))
+
+        if is_real_dtype(arr_out.dtype):
+            in_dtype = TYPE_MAP_R2C[arr_out.dtype]
+        elif halfcomplex:
+            raise ValueError('cannot combine halfcomplex backward transform '
+                             'with complex output')
+        else:
+            in_dtype = arr_out.dtype
+
+        if arr_in.dtype != in_dtype:
+            raise ValueError('expected input dtype {}, got {}'
+                             ''.format(dtype_repr(in_dtype),
+                                       dtype_repr(arr_in.dtype)))
+
+    else:  # Shouldn't happen
+        raise RuntimeError
```

Comment:
AFAIK the string in parentheses is optional, so with this you get the same exception without message. It's handled in an exception-specific way so you always get an instance.
```python
>>> e = RuntimeError
>>> print(e.args)
<attribute 'args' of 'BaseException' objects>
>>> try:
>>>     raise RuntimeError
>>> except RuntimeError as e:
>>>     print(e.args)
()
```